packages/block-library/src/audio/save.js

```diff
@@ -3,12 +3,12 @@
  */
 import { RichText } from '@wordpress/block-editor';
 
-export default function save( { attributes } ) {
+export default function save( { attributes, getBlockProps } ) {
```

Comment:
I'm not sure how I feel about this being a prop. Can't it be just a function you import for `@wordpress/blocks`. Also we need to think a bit ahead. What about inner blocks hook and rich text if we rework these instead, the list of props will keep growing.

Reply:
A hook could be nice, but we don't support it in save functions right? I don't immediately see another way to do this other than passing the block attributes and type/name to `getBlockProps`...

Reply:
Another thing you could do is set it as a property on save before calling it haha. `save.getBlockProps`.

Reply:
yes, maybe `getBlockProps( props )` 
Also, why do you consider this a blocker for the light block API? I mean the symmetry is good but there's no breaking change here right so it can be done at any moment?

Reply:
So

```js
function save( { attributes } ){
  return <div { ...save.getBlockProps( props ) }>{ attributes.text }</div>;
}

Reply:
Not sure if I like that more...

Reply:
yes, I feel like I'm not satisfied with any approach at the moment so I'm wondering if we can delay that to think properly. (the hook seems like a nice way, but we'd need a way to ensure that only sync hooks are allowed and we'd have to "fix" the renderer)

Reply:
> What about inner blocks hook and rich text if we rework these instead, the list of props will keep growing.

If we do the same for inner block, it might indeed need a function as well.
Rich text wouldn't need a function because it doesn't need context.

Reply:
@youknowriad Ok, that's fine. Let's delay it. We'll need to bump the version but that's fine.

Reply:
Oh, I have another idea. I'll change it now.

Reply:
@youknowriad Changed it to import from `@wordpress/blocks`. (Not sure if I like hooks for save functions. They're supposed to be simple while hooks are for behaviour. It's a lot of added complexity.)

Reply:
The last commit probably won't work for nested blocks. Would be nice to add a simple context system for the serialiser.

Reply:
Oh, it might work actually because the call order is top => down.

Reply:
So it pretty much works the same way as React context.